core/shared/src/main/scala/zio/ZEnvironment.scala

```diff
@@ -249,19 +252,24 @@ final class ZEnvironment[+R] private (
     private[ZEnvironment] def addService[A](tag: LightTypeTag, a: A)(implicit unsafe: Unsafe): ZEnvironment[R with A]
   }
 
+  trait UnsafeAPI4 {
+    private[zio] def getScope(implicit unsafe: Unsafe): Scope
```

Comment:
Given this is private, why can't we add it directly to avoid the unsafe API allocation?

Reply:
Good point, I'll add it directly to `ZEnvironment` instead

Reply:
Actually, I think we can make the method safe and public by requiring an `implicit ev: R <:< Scope`

Reply:
Sounds good